tests/cpp-tests/Classes/TerrainTest/TerrainTest.h

```diff
@@ -0,0 +1,91 @@
+#ifndef TERRAIN_TESH_H
+#include "../testBasic.h"
+#include "../BaseTest.h"
+
+#include "3d/CCSprite3D.h"
+#include "3d/CCTerrain.h"
+#include "2d/CCCamera.h"
+#include "2d/CCAction.h"
+class TerrainTestScene : public TestScene
+{
+public:
+    virtual void runThisTest();
+};
+
+class TerrainTestDemo : public BaseTest
+{
+public:
+    TerrainTestDemo(void);
+    virtual ~TerrainTestDemo(void);
+
+    void restartCallback(Ref* sender);
+    void nextCallback(Ref* sender);
+    void backCallback(Ref* sender);
+
+    // overrides
+    virtual std::string title() const override;
+    virtual std::string subtitle() const override;
+    virtual void onEnter() override;
+};
+
+
+class TerrainSimple : public TerrainTestDemo
+{
+public:
+    CREATE_FUNC(TerrainSimple);
+    TerrainSimple();
+    virtual std::string title() const override;
+    virtual std::string subtitle() const override;
+protected:
+};
+
+#define PLAYER_STATE_LEFT 0 
+#define PLAYER_STATE_RIGHT 1
+#define PLAYER_STATE_IDLE 2
+#define PLAYER_STATE_FORWARD 3
+#define PLAYER_STATE_BACKWARD 4
+
+class PlayerAction : public Action
```

Comment:
Action is confusing.

You can implement a class named Player, and there is a FSM in it.